%YAML:1.0

#####
##### Definitions
#####
# Input parameters are defined in Parameters.hpp
# Rotations are quaternions specified as: x,y,z,w (the same as ROS TF notation)


#####
##### Notes on this specific configuration file
#####
# Current camera settings for the robot.
# rosrun tf tf_echo base realsense_d435_forward_color_optical_frame


#####
##### Camera Parameters
##### works with anymal_2019-05-10-10-38-35.bag
cameras:
    - {
      image_dimensions: [848,480],
      focal_length:     [427.33, 426.79],
      focal_point:      [427.82, 238.32],

      # distortion_s:    0, # used by David but not used here
      distortion_k1:   0.0,
      distortion_k2:   0.0,
      distortion_k3:   0.0, # not used by David
      distortion_p1:   0.0,
      distortion_p2:   0.0,

      stereo_baseline: 0.05, # positive, as for David. internally fovis used -0.05

      B_r_BC: [0.0, 0.0, 0.0], # basic transform from camera optical (z fwd) to typical robotics (x fwd)
      B_q_BC: [0.5, -0.5, 0.5, -0.5]
      }
